Tools/CISupport/ews-build/steps_unittest.py

```diff
@@ -5738,6 +5743,60 @@ def test_failure(self):
         return self.runStep()
 
 
+class TestValidateRemote(BuildStepMixinAdditions, unittest.TestCase):
+    def setUp(self):
+        self.longMessage = True
+        return self.setUpBuildStep()
+
+    def tearDown(self):
+        return self.tearDownBuildStep()
+
+    def test_patch(self):
+        self.setupStep(ValidateRemote())
+        self.setProperty('patch_id', '1234')
+        self.expectOutcome(result=SKIPPED, state_string='finished (skipped)')
+        return self.runStep()
+
+    def test_origin(self):
+        self.setupStep(ValidateRemote())
+        self.setProperty('remote', 'origin')
+        self.expectOutcome(result=SKIPPED, state_string='finished (skipped)')
+        return self.runStep()
+
+    def test_success(self):
+        self.setupStep(ValidateRemote())
+        self.setProperty('remote', 'security')
+        self.setProperty('github.base.ref', 'safari-000-branch')
+        self.setProperty('github.number', '1234')
+        self.expectRemoteCommands(
+            ExpectShell(
+                workdir='wkdir',
+                logEnviron=False,
+                command=['git', 'merge-base', '--is-ancestor', 'remotes/security/safari-000-branch', 'remotes/origin/safari-000-branch'],
+            ) + 1,
+        )
+        self.expectOutcome(result=SUCCESS, state_string="Verified 'origin' does not own 'safari-000-branch'")
+        return self.runStep()
+
+    def test_failure(self):
+        self.setupStep(ValidateRemote())
+        self.setProperty('remote', 'security')
+        self.setProperty('github.base.ref', 'main')
+        self.setProperty('github.number', '1234')
+        self.expectRemoteCommands(
+            ExpectShell(
+                workdir='wkdir',
+                logEnviron=False,
+                command=['git', 'merge-base', '--is-ancestor', 'remotes/security/main', 'remotes/origin/main'],
+            ) + 0,
+        )
+        self.expectOutcome(result=FAILURE, state_string="Cannot land on 'main', it is owned by 'origin'")
```

Comment:
This error message seems very confusing. What does it mean to be owned by 'origin'. Whom should it be owned by?

Reply:
Maybe `WebKit/WebKit` is more accurate?

The branch should be owned by the remote which the PR is made against. So if I make a PR against `WebKit/WebKit-security`, it needs to be going to a branch owned by `WebKit/Webkit-security`.

Reply:
Yeah, `WebKit/WebKit` might be better. Also, would be better to include action which user can take. For example something like:

"Cannot land on 'main' since this branch is owned by WebKit/Webkit repo, should create PR agains XYZ"

Reply:
I'll add that to the PR comment.